lib/Path/Tiny.pm

```diff
@@ -247,11 +250,17 @@ sub path {
     }
 
     # do any tilde expansions
-    if ( $path =~ m{^(~[^/]*).*} ) {
+    if ($starts_with_tilde) {
         require File::Glob;
-        my ($homedir) = File::Glob::bsd_glob($1);
+        my ($tilde_part) = $path =~ m{^(~[^/]*)};
```

Comment:
Hmm. Is there any scenario where this match might fail? The rest of the block assumes it never will.